AspNetCore/AspNetCoreExtensions.cs

```diff
@@ -62,18 +83,69 @@ public static class AspNetCoreContextBuilderFactory
 		/// <param name="webHostBuilder">The ASP.NET Core web host builder from production code, overridden with mocks using .ConfigureTestServices().</param>
 		/// <param name="iocContainerFactory">Builds the IoC container, such as e.g. 'provider => new IocContainer(provider)'.</param>
 		public static void Initialize(Func<IWebHostBuilder> webHostBuilder, Func<IServiceProvider, IIocContainer> iocContainerFactory) =>
-		Initialize(CleanContextMode.ReCreate, webHostBuilder, iocContainerFactory);
+			Initialize(CleanContextMode.ReCreate, webHostBuilder, iocContainerFactory);
 
 		/// <summary>ASP.NET Core version of setting up IoC and builders to create the IoC context before each test.</summary>
 		/// <param name="hostBuilder">The ASP.NET Core host builder from production code, overridden with mocks using .ConfigureTestServices().</param>
 		/// <param name="iocContainerFactory">Builds the IoC container, such as e.g. 'provider => new IocContainer(provider)'.</param>
 		public static void Initialize(Func<IHostBuilder> hostBuilder, Func<IServiceProvider, IIocContainer> iocContainerFactory) =>
-		Initialize(CleanContextMode.ReCreate, hostBuilder, iocContainerFactory);
+			Initialize(CleanContextMode.ReCreate, hostBuilder, iocContainerFactory);
 
 		/// <summary>Get the ASP.NET Core <c>TestServer</c> created in <c>Initialize()</c>.</summary>
-		public static TestServer GetTestServer(this ContextBuilder _) => _testServer;
-
+		public static TestServer GetTestServer(this ContextBuilder _) => _wrapped?.GetTestServer();
 		/// <summary>Get the ASP.NET Core client from the <c>TestServer</c> created in <c>Initialize()</c>.</summary>
-		public static HttpClient GetClient(this ContextBuilder _) => _client;
+		public static HttpClient GetHttpClient(this ContextBuilder _) => _wrapped?.GetHttpClient();
+		/// <summary>Get the ASP.NET Core client from the factory created in <c>Initialize()</c>.</summary>
+		/// <remarks>Returns null if a Func&lt;WebApplicationFactory&lt;T&gt;&gt; was not passed in <c>Initialize()</c></remarks>
+		public static WebApplicationFactory<T> GetFactory<T>(this ContextBuilder _) where T: class => (_wrapped as FactoryWrapper<T>)?.GetFactory();
+
+		private interface IWrapper : IDisposable
```

Comment:
Indentation got a bit crazy from this part xD

Reply:
Right, let's use tabs & spaces consistently